src/vmod_cookie.c

```diff
@@ -213,6 +220,58 @@ vmod_get(VRT_CTX, struct vmod_priv *priv, VCL_STRING name)
 }
 
 
+vre_t *
+compile_re(VRT_CTX, VCL_STRING expression) {
+	vre_t *vre;
+	const char *error;
+	int erroroffset;
+
+	vre = VRE_compile(expression, 0, &error, &erroroffset);
+	if (vre == NULL) {
+		VSLb(ctx->vsl, SLT_VCL_Log, "cookie: PCRE compile error at char %i: %s", erroroffset, error);
+		return(NULL);
+	}
+	return(vre);
+}
+
+VCL_STRING
+vmod_get_re(VRT_CTX, struct vmod_priv *priv, struct vmod_priv *priv_call, VCL_STRING expression)
+{
+	struct vmod_cookie *vcp = cobj_get(priv);
+	(void)ctx;
+	int i, ovector[VRE_MAX_GROUPS];
+	struct cookie *cookie = NULL;
+	struct cookie *current;
+	vre_t *vre = NULL;
+
+	if (expression == NULL || *expression == '\0')
+		return(NULL);
+
+	if (priv_call->priv == NULL) {
+		vre = compile_re(ctx, expression);
+		if (!vre)
+			return(NULL);   // Not much else to do, error already logged.
+
+		priv_call->priv = vre;
+		priv_call->free = free;
+	}
+
+	VTAILQ_FOREACH(current, &vcp->cookielist, list) {
+		CHECK_OBJ_NOTNULL(current, VMOD_COOKIE_ENTRY_MAGIC);
+		VSLb(ctx->vsl, SLT_VCL_Log, "cookie: checking %s", current->name);
```

Comment:
would we really want to produce that much log traffic?
One option would be to have a debug flag in the vmod.